jsonnet/telemeter/rules.libsonnet

```
@@ -160,6 +160,51 @@
                 sum(sum_over_time(cluster:capacity_cpu_cores:sum{label_node_role_kubernetes_io = ''}[1h:5m])) by (_id) / scalar(count_over_time(vector(1)[1h:5m]))
               |||,
             },
+            {
+              // returns 0 for any cluster reporting core capacity, used to improve performance of cluster:capacity_effective_cpu_cores
+              record: 'cluster:cpu_capacity_cores:_id',
+              expr: |||
+                group by(_id) (node_role_os_version_machine:cpu_capacity_cores:sum) * 0
+              |||,
+            },
+            {
+              // OpenShift Cluster effective cores for subscription usage.
+              // This counts both worker nodes and, when the control plane is schedulable, control plane nodes.
+              // Only CoreOS nodes are counted.
+              // 1. x86_64 nodes that show hyperthreading in the telemetry have an accurate cores value in node_role_os_version_machine:cpu_capacity_cores:sum.
+              // 2. x86_64 nodes that do not show hyperthreading need the cores value adjusted to account for 2 threads per core (* 0.5).
+              // 3. Other CPU architectures are assumed to have accurate values in node_role_os_version_machine:cpu_capacity_cores:sum.
+              record: 'cluster:capacity_effective_cpu_cores',
+              expr: |||
+                # worker ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="true"}) or cluster:cpu_capacity_cores:_id)
+
+                +
+
+                # worker non-ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="false"}) / 2.0 or cluster:cpu_capacity_cores:_id)
+
+                +
+
+                # worker non-amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch!="amd64"}) or cluster:cpu_capacity_cores:_id)
+
+                +
+
+                # schedulable control plane ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_master="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="true"}) * on(_id) group by(_id) (cluster_master_schedulable) or cluster:cpu_capacity_cores:_id)
```

the current expression counts all cores of the master nodes even when cluster_master_schedulable is 0.

you need either (my preference)

```suggestion
                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_master="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="true"}) * on(_id) group by(_id) (cluster_master_schedulable == 1) or cluster:cpu_capacity_cores:_id)
```
or 

```suggestion
                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_master="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="true"}) * on(_id) max by(_id) (cluster_master_schedulable) or cluster:cpu_capacity_cores:_id)
```